packages/mobile/src/hooks/useScrollToTop.ts

```diff
@@ -0,0 +1,62 @@
+import { useCallback } from 'react'
+
+import {
+  NavigationProp,
+  useFocusEffect,
+  useNavigation
+} from '@react-navigation/native'
+
+/**
+ * A hook that listens for `scrollToTop` event on all parent navigators
+ * When the nearest navigator is type `tab`, listens to `tabPress` event
+ *
+ * react-navigation exports `useScrollToTop` but it doesn't support nested navigators
+ * see: https://github.com/react-navigation/react-navigation/issues/8586
+ */
+export const useScrollToTop = (
+  scrollToTop: () => void,
+  disableTopTabScroll = false
+) => {
+  const navigation = useNavigation()
+
+  useFocusEffect(
+    useCallback(() => {
```

Comment:
can we nest hooks like that?

Reply:
Yeah! The execution order isn't changing, this is essentially like assigning the result of `useCallback` to a a variable and passing that to `useFocusEffect` so it's ok

Reply:
ah word, yeah that makes sense